modules/local/format_prodigal.nf

```diff
@@ -20,9 +20,13 @@ process FORMAT_PRODIGAL {
     script:
     def args = task.ext.args   ?: ''
     prefix   = task.ext.prefix ?: "${meta.id}"
+    input    = gff =~ /\.gz$/ ? gff.name.take(gff.name.lastIndexOf('.')) : gff
+    gunzip   = gff =~ /\.gz$/ ? "gunzip -c ${gff} > ${input}" : ""
 
     """
-    sed 's/\\(\\(k[0-9]\\+_[0-9]\\+\\).*\\)ID=[0-9]\\+\\(_[0-9]\\+\\)/\\1ID=\\2\\3/g' $gff > ${prefix}_format.gff
+    $gunzip
+
+    sed 's/\\(\\(k[0-9]\\+_[0-9]\\+\\).*\\)ID=[0-9]\\+\\(_[0-9]\\+\\)/\\1ID=\\2\\3/g' $input > ${prefix}_format.gff
```

Comment:
I think it would be better to build one unix pipeline and avoid creating an unzipped file. E.g.:

```
cat_input = gff =~ /\.gz$/ ? "gunzip -c ${gff} : "cat ${gff}"

"""
$cat_input | sed 's/\\(\\(k[0-9]\\+_[0-9]\\+\\).*\\)ID=[0-9]\\+\\(_[0-9]\\+\\)/\\1ID=\\2\\3/g' | gzip -c > ${prefix}_format.gff.gz
"""
```

Above, I _guessed_ that one wants the output gzipped, but that is maybe not the case, or? If not, just remove the `| gzip -c`. *Edit* Now I see that you do want the output zipped, so keep the full pipeline and remove the last two lines, see comment below too.